lib/puppet/application/device.rb

```diff
@@ -169,8 +174,15 @@ def main
     # find device list
     require 'puppet/util/network_device/config'
     devices = Puppet::Util::NetworkDevice::Config.devices
+    if options[:node]
+      devices.select! { |key, value| key == options[:node] }
```

Comment:
I'm inclined to agree with @branan. 

A simpler solution than rewriting might just be to create a copy of `Puppet::Util::NetworkDevice::Config.devices` and then update later operations to operate on the copy variable?

I.e.,
```ruby
devices = Puppet::Util::NetworkDevice::Config.devices.dup
```
or
```ruby
selected_devices = devices.select { |key, value| key == options[:node] }
...
```